docs/source/nodes_and_pipelines/namespaces.md

```diff
@@ -164,3 +170,53 @@ def create_pipeline(**kwargs) -> Pipeline:
+
+For example, if you want to group the first two nodes of the `data_processing` pipeline from [Spaceflights tutorial](../tutorial/add_another_pipeline.md#data-science-pipeline) into the same namespace, you can do:
+
+```python
+#src/project_name/pipelines/data_science/pipeline.py
+
+from kedro.pipeline import Pipeline, node, pipeline
+from .nodes import create_model_input_table, preprocess_companies, preprocess_shuttles
+
+def create_pipeline(**kwargs) -> Pipeline:
+    return pipeline(
+        [
+            node(
+                func=preprocess_companies,
+                inputs="companies",
+                outputs="preprocessed_companies",
+                name="preprocess_companies_node",
+                namespace="preprocessing", # Assigning the node to the "preprocessing" namespace
+            ),
+            node(
+                func=preprocess_shuttles,
+                inputs="shuttles",
+                outputs="preprocessed_shuttles",
+                name="preprocess_shuttles_node",
+                namespace="preprocessing", # Assigning the node to the "preprocessing" namespace
+            ),
+            node(
+                func=create_model_input_table,
+                inputs=["preprocessed_shuttles", "preprocessed_companies", "reviews"],
+                outputs="model_input_table",
+                name="create_model_input_table_node",
+            ),
+        ]
+    )
+```
+
+As you can see in the example above, we are using the `namespace` option only in the first two node creation functions. In this case, only the node names themselves will be prefixed with `preprocessing.`, while the inputs, outputs, and parameters remain unchanged.
+
+Whether the namespace is assigned at the pipeline or node level, you can execute it separately using:
+```bash
+kedro run --namespace=<your_namespace>
+```
+
+Additionally, Kedro-Viz will allow you to expand and collapse your namespace for better visualisation.
+
+> **Caution:** While this functionality is useful for grouping nodes easily without additional complexities that may arise when creating a fully namespaced pipeline, be aware that it can create a namespace that is **non-executable**. This happens if you group, for example, the first and last nodes of a pipeline without including the intermediate nodes in the same namespace. In such cases, attempting to execute that namespace with `kedro run --namespace=<your_namespace>` will result in an error, and your visualisation in Kedro-Viz will be broken.
```

Comment:
Can this part be in a warning notation: like this https://github.com/kedro-org/kedro/blob/7b4be6b53e70788f5af3c55d01b60280f3d9064f/docs/source/data/advanced_data_catalog_usage.md?plain=1#L5

Reply:
There's some more sections in this page which are not modified in this PR which are formatted like this which could be formatted as ```{note} sections?